### PR TITLE
AddExclusionListToPolicy

### PR DESCRIPTION
**Pull request by BeckyHope19**
## Overview/Summary

The pull request being submitted is for a policy update to Deny vNet peering to other subscriptions, this update includes a new parameter to allow for vnets in other subscriptions.

## This PR fixes/adds/changes/removes

1. This change adds a parameter field to exclude certain vNets from being denied to peer with vNets in different subscriptions.

### Breaking Changes

1. N/A

## Testing Evidence

 I carried out tests in my test subscription with 4 vnets, 2 were in the same subscription and the other 2 were in another subscription. vNet 1 & 3 sit in Subscription 1 and vNet 2 & 4 sit in Subscription 2. Both subscription sit under the bex19 management group scope. 

Here you will see the policy has been assigned to the bex19 management group scope, this is where the vnets also sit.
PolicyAssigned

You can then see the parameters I have passed in the policy for vNet 1 & 2 to be allowed to peer, you can see the subscription ID is different in the screenshot:
PolicyParameters

Here I am trying to create a peering between vNet 3 & 4, you will see it fails as neither of the vNets are in the allowed list and they are both in different subscriptions:
vnet 3-4 failed peering

Here I have created a peering between vNet 1 & 3, you will see it is successful as they are both in the same subscription:
vnet 1-3 successful peering

Here I have created a peering between vNet 2 & 4, you will see it is successful as they are both in the same subscription:
vnet 2-4 successful peering

Here I have created a peering between vNet 1 & 2, you will see it is successful as they are both in the allowed list:
vnet 1-2 successful peering

Here I am trying to create a peering between vNet 1 & 4, you will see it fails because only vNet 1 is in the allowed list and vNet 4 isn't, so it is only allowing the peering one way:
vnet 1-4 failed peering

The below URLs can be updated where the placeholders are, look for `BeckysBranchALZ-07-08-24` & `BeckysBranchALZ-07-08-24`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FBeckyHope19%2FEnterprise-Scale%2FBeckysBranchALZ-07-08-24%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FBeckyHope19%2FEnterprise-Scale%2FBeckysBranchALZ-07-08-24%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [ ] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by BeckyHope19** [comment]
@Springstone please fix this for me :-)

File: src/resources/Microsoft.Authorization/policyDefinitions/Deny-VNET-Peer-Cross-Sub.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "displayName": "Deny vNet peering cross subscription.",
     "description": "This policy denies the creation of vNet Peerings outside of the same subscriptions under the assigned scope.",
     "metadata": {
-      "version": "1.0.1",
+      "version": "1.0.2",
```
Comment (Springstone) [inline src/resources/Microsoft.Authorization/policyDefinitions/Deny-VNET-Peer-Cross-Sub.json:12]: As we are adding a new parameter, we need to increment the minor version of the policy as per https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Policies#versioning.
Please change policy version to "1.1.0".

**Review by Springstone — CHANGES_REQUESTED** [review]
Great contribution. Couple of small things, and please can you also update "Whatsnew.md" in the wiki under the Policy Refresh section to include a short description on what you changed. 
Once you've changed the policy version, please re-run the policy build script.

**Comment by BeckyHope19** [comment]
@Springstone - I have added the changes to the parameters as requested and updated the Whats New page and run the build command again.

**Review by Springstone — APPROVED** [review]

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**